### PR TITLE
CostModelContainer: Make CostItem a shared_ptr

### PR DESCRIPTION
**Pull request by wxmerkt**
Related to #672

### CODE REVIEW

**Comment by wxmerkt** [comment]
I need to fix the Python bindings. Unfortunately, I am not that experienced with Boost.Python as I mostly use Pybind11 that does these translations automatically. I will provide a fix later this afternoon.

**Comment by wxmerkt** [comment]
I still get this error:

```
Traceback (most recent call last):
  File "unittest/bindings/test_momemtum_cost.py", line 68, in <module>
    cmodel = model.costs.costs['momentum'].cost
TypeError: No Python class registered for C++ class boost::shared_ptr<crocoddyl::CostItemTpl<double> >
```

Alas, I don't know what I need to do in Boost.Python beyond `register_ptr_to_python`. Can you please advise? :-)

File: bindings/python/crocoddyl/multibody/costs/cost-sum.cpp [hunk 1]
```diff
@@ -24,14 +24,17 @@ BOOST_PYTHON_MEMBER_FUNCTION_OVERLOADS(CostModelSum_calc_wraps, CostModelSum::ca
...
+  typedef boost::shared_ptr<CostItem> CostItemPtr;
   typedef boost::shared_ptr<CostDataAbstract> CostDataPtr;
-  bp::to_python_converter<std::map<std::string, CostItem, std::less<std::string>,
-                                   std::allocator<std::pair<const std::string, CostItem> > >,
-                          map_to_dict<std::string, CostItem> >();
+
+  bp::register_ptr_to_python<boost::shared_ptr<CostItem> >();
+  bp::to_python_converter<std::map<std::string, CostItemPtr, std::less<std::string>,
+                                   std::allocator<std::pair<const std::string, CostItemPtr> > >,
+                          map_to_dict<std::string, CostItemPtr> >();
```
Comment (cmastalli) [inline bindings/python/crocoddyl/multibody/costs/cost-sum.cpp]: I think you need to add the `false` flag, i.e.
```cpp
  bp::to_python_converter<std::map<std::string, CostItemPtr, std::less<std::string>,
                                   std::allocator<std::pair<const std::string, CostItemPtr> > >,
                          map_to_dict<std::string, CostItemPtr, false> >();
```

This should fix the issue. Let me know if this is not the case

Comment (wxmerkt) [reply bindings/python/crocoddyl/multibody/costs/cost-sum.cpp]: Thank you, this fixed it! I pushed the changes now and it should pass the buildserver soon.

**Comment by wxmerkt** [comment]
Thank you, this is now passing CI and should be good to merge once Travis goes green :-).

**Comment by cmastalli** [comment]
> Thank you, this is now passing CI and should be good to merge once Travis goes green :-).

Cool. Thanks for making this effort :)

**Pull request merged**